patches/minecraft/net/minecraft/entity/ai/brain/task/ChangeJobTask.java.patch

```diff
@@ -0,0 +1,12 @@
+--- a/net/minecraft/entity/ai/brain/task/ChangeJobTask.java
++++ b/net/minecraft/entity/ai/brain/task/ChangeJobTask.java
+@@ -19,7 +19,9 @@
+    }
+ 
+    protected void func_212831_a_(ServerWorld p_212831_1_, VillagerEntity p_212831_2_, long p_212831_3_) {
++      VillagerProfession oldProfession = p_212831_2_.func_213700_eh().func_221130_b();
+       p_212831_2_.func_213753_a(p_212831_2_.func_213700_eh().func_221126_a(VillagerProfession.field_221151_a));
+       p_212831_2_.func_213770_a(p_212831_1_);
++      net.minecraftforge.event.ForgeEventFactory.onVillagerProfessionChange(p_212831_2_, oldProfession);
```

Comment:
Why are these called from the tasks? There are other paths that change the profession.
Why not just call it from setVillagerData, which both these hooks call?

Reply:
The idea behind adding them to the tasks was so they could fire after the brain reset, at which point the profession change feels more "complete". But I see your point about other code paths potentially changing things, so the event is now in `setVillagerData`.